src/services/createOrUpdateGoalsForActivityReport.test.js

```diff
@@ -0,0 +1,411 @@
+      activityReportId: otherEntityReport.id,
+      otherEntityId: otherEntity.id,
+    });
+  });
+
+  afterAll(async () => {
+    // Delete ARO.
+    await ActivityReportObjective.destroy({
+      where: {
+        activityReportId: activityReport.id,
+      },
+    });
+
+    // Delete ARG.
+    await ActivityReportGoal.destroy({
+      where: {
+        activityReportId: activityReport.id,
+      },
+    });
+
+    // Delete Recipient AR.
+    await ActivityReport.destroy({ where: { id: activityReport.id } });
+
+    // Delete OE ARO.
+    await ActivityReportObjective.destroy({
+      where: {
+        activityReportId: otherEntityReport.id,
+      },
+    });
+
+    // Delete OE Obj's.
+    await Objective.destroy({ where: { id: oeObjectiveIds } });
+
+    // Delete OE Recipient.
+    await ActivityRecipient.destroy({
+      where: {
+        activityReportId: otherEntityReport.id,
+      },
+    });
+
+    // Delete OE AR.
+    await ActivityReport.destroy({ where: { id: otherEntityReport.id } });
+
+    // Delete Recipient Obj's
+    await Objective.destroy({ where: { goalId: goalIds } });
+
+    // Delete Goal.
+    await Goal.destroy({
+      where: {
+        id: goalIds,
+      },
+    });
+
+    // Delete Grant.
+    await Grant.destroy({
+      where: {
+        id: grants.map((g) => g.id),
+      },
+    });
+
+    // Delete Recipient.
+    await Recipient.destroy({
+      where: {
+        id: recipient.id,
+      },
+    });
+
+    // Delete OE.
+    await OtherEntity.destroy({
+      where: {
+        id: otherEntity.id,
+      },
+    });
+
+    // Delete User.
+    await User.destroy({
+      where: {
+        id: user.id,
+      },
+    });
+
+    // Close Conn.
+    await db.sequelize.close();
+  });
+
+  it('creates other entity new objectives and updates existing ones', async () => {
+    const objectivesToCreate = [
+      {
+        title: 'Test create OE objective - Obj 1',
+        status: 'Not Started',
+        recipientIds: [otherEntity.dataValues.id],
+        ttaProvided: '<p>Test create OE objective - Obj 1 tta</p>',
+        topics: [],
+        resources: [],
+        files: [],
+        isNew: true,
+      },
+      {
+        title: 'Test create OE objective - Obj 2',
+        status: 'In Progress',
+        recipientIds: [otherEntity.dataValues.id],
+        ttaProvided: '<p>Test create OE objective - Obj 2 tta</p>',
+        topics: [],
+        resources: [],
+        files: [],
+        isNew: true,
+      },
+      {
+        title: 'Test create OE objective - Obj 3',
+        status: 'Complete',
+        recipientIds: [otherEntity.dataValues.id],
+        ttaProvided: '<p>Test create OE objective - Obj 3 tta</p>',
+        topics: [],
+        resources: [],
+        files: [],
+        isNew: true,
+      },
+    ];
+
+    await saveObjectivesForReport(
+      objectivesToCreate,
+      otherEntityReport,
+    );
+
+    const createdObjectives = await getObjectivesByReportId(otherEntityReport.id);
```

Comment:
It seems the function 'createOrUpdateGoalsForActivityReport()' doesn't support OE reports. So I added a CRUD test here with 'saveObjectivesForReport()'.